modules/currency-l1/src/main/scala/org/tessellation/currency/l1/Main.scala

```diff
@@ -0,0 +1,177 @@
+          sdkServices.session,
+          cfg.healthCheck,
+          sdk.nodeId
+        )
+        .asResource
+
+      rumorHandler = RumorHandlers.make[IO](currencyStorages.cluster, healthChecks.ping, services.localHealthcheck).handlers <+>
+        blockRumorHandler[IO, CurrencyBlock](queues.peerBlock)
+
+      _ <- Daemons
+        .start(currencyStorages, services, healthChecks)
+        .asResource
+
+      api = HttpApi
+        .make[IO, CurrencyTransaction, CurrencyBlock, CurrencySnapshot](
+          currencyStorages,
+          queues,
+          keyPair.getPrivate,
+          services,
+          programs,
+          healthChecks,
+          sdk.nodeId,
+          BuildInfo.version,
+          cfg.http
+        )
+      _ <- MkHttpServer[IO].newEmber(ServerName("public"), cfg.http.publicHttp, api.publicApp)
+      _ <- MkHttpServer[IO].newEmber(ServerName("p2p"), cfg.http.p2pHttp, api.p2pApp)
+      _ <- MkHttpServer[IO].newEmber(ServerName("cli"), cfg.http.cliHttp, api.cliApp)
+
+      stateChannel <- StateChannel
+        .make[IO, CurrencyTransaction, CurrencyBlock, CurrencySnapshot](
+          cfg,
+          keyPair,
+          p2pClient,
+          programs,
+          queues,
+          nodeId,
+          services,
+          currencyStorages,
+          validators
+        )
+        .asResource
+
+      gossipDaemon = GossipDaemon.make[IO](
+        currencyStorages.rumor,
+        queues.rumor,
+        currencyStorages.cluster,
+        p2pClient.gossip,
+        rumorHandler,
+        validators.rumorValidator,
+        services.localHealthcheck,
+        nodeId,
+        generation,
+        cfg.gossip.daemon,
+        services.collateral
+      )
+      _ <- {
+        method match {
+          case cfg: RunInitialValidator =>
+            gossipDaemon.startAsInitialValidator >>
+              programs.l0PeerDiscovery.discoverFrom(cfg.l0Peer) >>
+              programs.globalL0PeerDiscovery.discoverFrom(cfg.globalL0Peer) >>
+              currencyStorages.node.tryModifyState(NodeState.Initial, NodeState.ReadyToJoin) >>
+              services.cluster.createSession >>
+              services.session.createSession >>
+              currencyStorages.node.tryModifyState(SessionStarted, NodeState.Ready)
+
+          case cfg: RunValidator =>
+            gossipDaemon.startAsRegularValidator >>
+              programs.l0PeerDiscovery.discoverFrom(cfg.l0Peer) >>
+              programs.globalL0PeerDiscovery.discoverFrom(cfg.globalL0Peer) >>
+              currencyStorages.node.tryModifyState(NodeState.Initial, NodeState.ReadyToJoin)
+        }
+      }.asResource
+      globalL0PeerDiscovery = Stream
+        .awakeEvery[IO](10.seconds)
+        .evalMap { _ =>
+          currencyStorages.lastGlobalSnapshot.get.flatMap {
+            _.fold(Applicative[IO].unit) { latestSnapshot =>
+              programs.globalL0PeerDiscovery.discover(latestSnapshot.signed.proofs.map(_.id).map(PeerId._Id.reverseGet))
```

Comment:
I have a feeling that it should be Daemon/Program.

Reply:
@TheMMaciek shouldn't it be a part of state channel runtime?

Reply:
It could be a daemon, it could also be a part of the state channel class. But for now I would consider that this solution is at least ok. In a way it is a part of the state channel runtime because I merge both streams here. I just wanted to avoid premature abstracting of state channel class functionality (or injecting with a parameter) when this part is the only part I needed to add. I would revisit it the moment we will tackle the extendability of L0 token functionality as we may then decide on how the custom stuff should be added to the implementation and hence also this part could fall into place then.